vanilla/vanilla
# Add VanillaConnect addon

Requires https://github.com/vanilla/vanilla/pull/5831

# How to test

- Clone https://github.com/vanilla/stub-sso-providers/pull/1
    - Composer install
- Setup nginx to work with stub-sso-providers
    - Base yourself on https://github.com/vanilla/vanilla-docker/pull/1/files#diff-4e5487559f07f86005bca108c798464e
- Create a config.php in stub-sso-providers (The [template work as is](https://github.com/vanilla/stub-sso-providers/pull/1/files#diff-ce65fe62e6de6a262dc6fefc3c869e7f) in the current PR if your host is vanilla.dev)
- Enable the plugin
- Add a provider ![screen shot 2017-10-04 at 12 13 23 pm](https://user-images.githubusercontent.com/2412909/31186728-7917dd48-a8fd-11e7-9889-6489a236269f.png)
- Read this: https://github.com/vanilla/vanilla-connect-php/blob/b006096d5fc011c1343860aa1aa32d0ade36eea1/README.md#vanillaconnect-library-for-php
- From there you need to make sure that #5831 is on your branch.
- You need to edit [this file](https://github.com/vanilla/stub-sso-providers/pull/1/files#diff-1c7f01aedb48917745244fe08e0c3f68R10) to change the content of the response JWT's claim.

# What to test

- [ ] Default vs non default connection.
    - [ ] If set as default hitting /sso should try to log you in.
- [ ] Custom redirection using the `target` parameter. 
    - Example: `http://sso.dev/vanillaconnect/signin?target=/discussions`
- User information synchronization. Refer to VanillaConnect [claim section](https://github.com/vanilla/vanilla-connect-php/blob/b006096d5fc011c1343860aa1aa32d0ade36eea1/README.md#claim)
    - [ ] name, email, photourl
- [ ] URL not properly whitelisted
- [ ] [Push SSO](https://github.com/vanilla/stub-sso-providers/pull/1/files#diff-087ead604a162f232637899477accd5e)
